### DISCUSSION TITLE
docs/web/http-client-file-uploading

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#54] [General]
# docs/web/http-client-file-uploading

使用GoFrame框架进行HTTP客户端文件上传，实现了方便的文件上传功能，并提供了三个主要的接口以支持单个和多个文件的上传。详细讲解了服务端及客户端的实现代码，并提供了自定义文件名称和规范路由接收上传文件的方法，适用于需要集成文件上传功能的开发场景。

https://goframe.org/docs/web/http-client-file-uploading

**Comment by IshwarShrivastav** [comment]
通过g.Client()获取了一个文件  如何转为ghttp.UploadFile？

**Reply by IshwarShrivastav** [reply]
ghttp.UploadFile只能通过前端传过来吗？
后端自己请求文件 能不能转为ghttp.UploadFile？

**Reply by shiqinfeng1** [reply]
报错：  reflect.Value.Convert: value of type string cannot be converted to type ghttp.UploadFile

**Comment by DustScribe** [comment]
如果使用 `Apifox` 的话，只有 `type:"file"` 接口文档不会变成 `file` 类型，添加 `type:"file" format:"binary"` 后就可以识别为 `file` 类型了

**Comment by ahcheqiu** [comment]
对于大文件上传似乎需要等到服务器把文件完全下载完成才会进入到第一行代码，想要流式处理大文件似乎不行？
